app/services/fetch_coronavirus_statistics_service.rb

```diff
@@ -113,12 +114,36 @@ def latest_admissions(data)
 
   def latest_tests(data)
     latest_tests = data.find { |d| d["newPositiveTests"] }
-
     return {} unless latest_tests
 
+    daily_tests = daily_tests(latest_tests)
+    weekly_tests = weekly_tests(latest_tests, data)
+
+    return {} unless daily_tests && weekly_tests
+
+    daily_tests.merge!(weekly_tests)
+  end
+
+  def daily_tests(latest_tests)
     {
       new_positive_tests: latest_tests["newPositiveTests"],
       new_positive_tests_date: Date.parse(latest_tests["date"]),
     }
   end
+
+  def weekly_tests(latest_tests, data)
+    day_tests = []
+    data.each do |day_cases|
+      day_tests << day_cases["newPositiveTests"] if latest_tests["date"] >= day_cases["date"]
+    end
+
+    current_week_tests = day_tests.first(7).compact
```

Comment:
Probably just me missing something, but I am confused about how the latest values with `nil` records are ignored (as demonstrated in the commit message). 

`a = [nil, 2, 3, 4, 5, 6, 7, 8, 9, 10]` 

` b = a.first(7).compact`  

`=> [2, 3, 4, 5, 6, 7]`

`b.count`

`=> 6` 

which would `return` on the line below

Reply:
So, in you example, if `a` or `data` is `[nil, 2, 3, 4, 5, 6, 7, 8, 9, 10]`
Then `day_tests` is populated with `[2, 3, 4, 5, 6, 7, 8, 9, 10]`

```
b = day_tests.first(7).compact
=> [2, 3, 4, 5, 6, 7, 8]

Reply:
`latest_tests["date"]` is the first hash in the data had has a value for number of positive tests, so thats when we start recording the value.

Reply:
ah ok! Looking at the data in the tests it looks like it is possible for the `date` value to be present but then the other fields in that chunk are `nil`. So I wondered if `day_cases["newPositiveTests"]` would be `nil` which gets stored in `day_tests`? Which could mean the `nil` would get stored and effect the count?

Reply:
ah sorry just spoke to Jon and realised I misunderstood. If the first value is nil then we would want to `return` as you mentioned in the commit message 'as this proves that we don't have the full 7 days worth of data and should not display the section'.

I incorrectly thought we would want to remove the nils at the start, but keep counting from the place where we had data and display that, whereas actually we don't want to display anything due to missing data.

Reply:
@Rosa-Fox Is there anything I can add to the commit message to make this clearer?

Reply:
You're right. If we have nils at the start we want to remove them. And start our "7 days" in the past. But if there is nil data in the middle of a week, we want to `return` as we won't have a full week's worth of data.